routes/user/index.js

```diff
@@ -0,0 +1,55 @@
+const express = require( 'express' )
+const router = express.Router()
+const passport = require( '../../configuration/passport' )
+const User = require( '../../database/user' )
+
+router.post( '/signup', ( request, response, next ) => {
+  const { email, password } = request.body
+
+  User.findByEmail( email )
+    .then( user => {
+      request.flash( 'message', 'Email already registered.' )
+      response.redirect( '/' )
+    })
+    .catch( error => {
+      const { hash, salt } = User.generateHash( password )
+
+      User.create( email, hash, salt )
+        .then( user => {
+          request.login( user, error => {
+            if( error ) {
+              return next( error )
+            } else {
+              response.redirect( '/users/profile' )
+            }
+          })
+        })
+        .catch( error => {
+          console.log( 'error creating user', error )
+        })
+    })
+})
+
+const authConfig = {
```

Comment:
If this is a constant value, please declare it at the top of the file (before all routes). That way it's easier to see the separation in the file between defining values and declaring routes.